src/core/lib/channel/channelz_registry.h

```diff
@@ -82,16 +83,20 @@ class ChannelzRegistry {
   template <typename Object>
   Object* InternalGet(intptr_t uuid) {
     gpr_mu_lock(&mu_);
-    Object* ret =
-        static_cast<Object*>(grpc_avl_get(avl_, (void*)uuid, nullptr));
+    if (uuid < 1 || uuid > static_cast<intptr_t>(entities_.size())) {
+      gpr_mu_unlock(&mu_);
+      return nullptr;
+    }
+    Object* ret = static_cast<Object*>(entities_[uuid - 1]);
     gpr_mu_unlock(&mu_);
     return ret;
   }
 
   // private members
+
+  // protects entities_ and uuid_
   gpr_mu mu_;
-  grpc_avl avl_;
-  gpr_atm uuid_;
```

Comment:
Unrelated to this PR, but I've just noticed that in the current implementation, there's probably no reason for this to be an atm.  We're taking a lock to modify the avl anyway, so we can probably just make this an int and manipulate it inside the lock.

Reply:
I will make this change if we revert to avl